chia/wallet/coin_selection.py

```diff
@@ -0,0 +1,145 @@
+import random
+from typing import Set, Optional, List, Dict
+
+from chia.types.blockchain_format.coin import Coin
+from chia.types.blockchain_format.sized_bytes import bytes32
+from chia.util.ints import uint64
+from chia.wallet.wallet_coin_record import WalletCoinRecord
+
+
+async def select_coins(wallet, amount: uint64, exclude: List[Coin] = None) -> Set[Coin]:

```

Comment:
It's not good practice to pass the larger higher level object here (wallet).

Reply:
Instead, we can pass in the required data:
- spendable cons
- constants
- unconfirmed removals